Update ansible role

mbstacy opened the pull request:
I have updated role to install the latest version of docker. Current docker stable version is 1.11. Other update was to configure docker to use overlay storage option. The default out of the box for Centos distribution  is the devicemapper with loopback file storage configuration, which is not recommended for production systems.

jsnshrmn commented:
Mark, thanks for pitching in!

This is a good change to use the official docker repo instead of the slow to update centos provided repo. The distro-provided repo will only fall further behind. Thanks for moving us to a better storage configuration too.

A few thoughts:

Instead of copying over a yum repo file, you should use the native yum support to install the docker repo.  See an example here:
https://github.com/OULibraries/ansible-role-zabbix-agent/blob/master/tasks/yum.yml

The bit where you are adding the vagrant user to the docker group should be done in the vagrant_docker repo rather than this role:
https://github.com/OULibraries/vagrant_docker

When you want to have something run at the end, you can set up a handler. Handlers run after all tasks complete.  You can do setup a check in vagrant_docker to see if vagrant is a member of the docker group, and notify a handler to add the user to the group if it isn't there already.

You should also note that vagrant has native support for managing users and groups, which is less error-prone than running the shell commands. Less of a big deal on a vagrant install than on a role that gets applied to production.

After making these changes, only service configuration changes will be left in tasks/files.yml, so it should be renamed tasks/configure.yml to be consistent with our other roles.

tasks/final.yml will go away since those bits will be handled in vagrant_docker

Let me know if you have questions or if you would like me to tackle any of these suggested changes!

mbstacy commented:
Jason,
Thanks for the instructions. I believe that I have fixed all issues: configure.yml; yum.yml to add repo through ansible and removed the vagrant user docker group. I have not updated the vagrant_docker repo, but will shortly to add vagrant user to docker group. Not sure about handlers, but will ask if run into problems.
